src/builder.rs

```diff
@@ -0,0 +1,213 @@
+use crate::{
+    private::{RedactFlags, RedactionFormatter, RedactionTarget},
+    util::give_me_a_formatter,
+};
+use std::fmt::{Debug, Display};
+
+/// A wrapped reference to some data that, when formatted as [`Debug`] or [`Display`] (if implemented for `T`), will be redacted.
+///
+/// See [`Redactor::wrap`] for more information.
+pub struct RedactWrapped<'a, T> {
+    data: &'a T,
+    flags: &'a RedactFlags,
+}
+impl<T> Display for RedactWrapped<'_, T>
+where
+    T: Display,
+{
+    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Debug::fmt(
+            &RedactionFormatter {
+                this: RedactionTarget::Display(self.data),
+                flags: *self.flags,
+                specialization: None,
+            },
+            fmt,
+        )
+    }
+}
+impl<T> Debug for RedactWrapped<'_, T>
+where
+    T: Debug,
+{
+    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Debug::fmt(
+            &RedactionFormatter {
+                this: RedactionTarget::Debug {
+                    this: self.data,
+                    alternate: fmt.alternate(),
+                },
+                flags: *self.flags,
+                specialization: None,
+            },
+            fmt,
+        )
+    }
+}
+impl<T> Copy for RedactWrapped<'_, T> {}
+impl<T> Clone for RedactWrapped<'_, T> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+/// The `Redactor` allows for redacting arbitrary strings using a pre-defined set of flags.
+///
+/// To build a `Redactor`, use the [`RedactorBuilder`].
+pub struct Redactor(RedactFlags);
+impl Redactor {
```

Comment:
I would suggest adding a "builder" associated function here that returns a RedactorBuilder